### PR TITLE
[New] Add reframe.js@2.1.18 w/ npm auto-update

### LINKED ISSUE #10072: [Request] Add reframe.js
**Library name:** reframe.js
**Git repository url:** https://github.com/dollarshaveclub/reframe.js
**npm package url(optional):** https://www.npmjs.com/package/reframe.js
**License(s):** MIT
**Official homepage:** https://dollarshaveclub.github.io/reframe.js/
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sashberd**
close #10072, cc @dollarshaveclub
Pull request for issue: #10072
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/dollarshaveclub/reframe.js
 * Official website (optional, not the repository): https://dollarshaveclub.github.io/reframe.js/
 * NPM package url (optional): https://www.npmjs.com/package/reframe.js
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 42
   - Count of stars: 1375
   - Count of forks: 50
 * NPM download stats (optional):
   - Downloads in the last day: 124
   - Downloads in the last week: 293
   - Downloads in the last month: 1320

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/reframe.js/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "padding-top"
+  ],
+  "author": "Jeff Wainwright <jjwainwright2@gmail.com> (http://jeffry.in)",
+  "homepage": "https://dollarshaveclub.github.io/reframe.js/",
+  "npmName": "reframe.js",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*.+(js|css)"
```
Comment (extend1994) [inline ajax/libs/reframe.js/package.json]: Let's use `**/!(*.scss)` here. In kind of `dist` folders, cdnjs would like to be more flexible to fetch files. So usually excluding files would be better choice.

Please leave a comment in PR to let reviewers know why you use some special rules to speed up review. Thanks.

File: ajax/libs/reframe.js/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "iframes",
+    "fluid",
+    "responsive",
+    "video",
+    "fitvids",
+    "youtube",
+    "embeds",
+    "padding-top"
+  ],
+  "author": "Jeff Wainwright <jjwainwright2@gmail.com> (http://jeffry.in)",
```
Comment (extend1994) [inline ajax/libs/reframe.js/package.json]: The url can be `https`.

**Comment by extend1994** [comment]
Forget to mention that when the repo belongs to an organization, you should cc main contributor/author/maintainers depends on the project. Because if you cc an organization, no one will be notified. In this case, you can `cc @yowainwright`

**Comment by extend1994** [comment]
Forget to mention another thing :sweat_smile: 
The 2nd commit msg `added from git` part, please use `added from git **repo**`. git is a version control tool, you won't be able to add from it.

**Comment by sashberd** [comment]
@extend1994 
The PR was updated. FYR
About the commits messages:
Guys all of you have different requirements about the commit messages. I so confused what type of commit message I should use. Any time  of multiple commit, each of you provide me different reviews about how commit message should be looked. 
I do not have any problem to change any of my commit message, but it is taking to much time from you ,reviewrs, and me ,contributor, to take care for commit messages.
@PeterDaveHello Do we have any documentation about how first, second, third... commit should look. If we do not, please open issue to add such documentation.
Thanks

**Comment by extend1994** [comment]
@sashberd I think the commit messages we mentioned in you pull requests are all "suggestions". You don't need to use the totally same commit msg as we suggested. 
1. cc an user rather than an organization are general rule on GitHub.
2. I suggested you to leave a **comment** about the pattern of auto-update in **pull request** instead of "commit message".
3. "from git **repo**" makes more sense than "git"

The aboves are not directly related "cdnjs commit message rule"

**Comment by sufuf3** [comment]
Agree with @extend1994 
@sashberd would you modify the commit?
I think you can refer to the log of commits, these commit message will help you write a good commit message.
Also in second commit message, about `exsitnence` which I don't know this word not sure it is right.
About add old versions' detail commit message, I'd like to use `Add vx.x.x from GitHub because ...`, but using `from git repository` or `from git repo` also looks good to me.
`from git` is a little bit weird because git is a tool not repo.

Would you mind modifying the commit message to improve this PR also help this project better in every commit?

Many thanks.

**Comment by sashberd** [comment]
@sufuf3 @extend1994 The PR message was updated. FYR

**Comment by sashberd** [comment]
@sufuf3 The Pr commit message as updated. FYR

**Comment by yowainwright** [comment]
@sashberd thanks for adding this!!!

Team, I approve adding **reframe.js**!

~Thank you

**Comment by sashberd** [comment]
@sufuf3 ping

File: ajax/libs/reframe.js/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "padding-top"
+  ],
+  "author": "Jeff Wainwright <jjwainwright2@gmail.com> (https://jeffry.in)",
+  "homepage": "https://dollarshaveclub.github.io/reframe.js/",
+  "npmName": "reframe.js",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/!(*.scss)"
```
Comment (PeterDaveHello) [inline ajax/libs/reframe.js/package.json]: I guess `noframe.es.js` and `reframe.es.js` need to be excluded?

**Comment by sashberd** [comment]
@PeterDaveHello The PR was updated. FYR

File: ajax/libs/reframe.js/package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    "padding-top"
+  ],
+  "author": "Jeff Wainwright <jjwainwright2@gmail.com> (https://jeffry.in)",
+  "homepage": "https://dollarshaveclub.github.io/reframe.js/",
+  "npmName": "reframe.js",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/!(*.scss || *.es.js)"
```
Comment (PeterDaveHello) [inline ajax/libs/reframe.js/package.json]: `"**/!(*.scss || *.es.js)"` not `"**/!(*.scss|*.es.js)"`, would you please test the rules locally?

**Comment by PeterDaveHello** [comment]
LGTM now

**Pull request merged**